yearn/contracts/SaddleStrategy.sol

```diff
@@ -0,0 +1,395 @@
+    function balanceOfWant() public view returns (uint256) {
+        return want.balanceOf(address(this));
+    }
+
+    /// @return Balance of the vault's underlying token staked into the tBTC v2
+    ///         Saddle reward pool.
+    function balanceOfPool() public view returns (uint256) {
+        return
+            ILPRewardsTBTCV2Saddle(lpRewardsTbtcV2Saddle).balanceOf(
+                address(this)
+            );
+    }
+
+    /// @return Sum of balanceOfWant and balanceOfPool.
+    function estimatedTotalAssets() public view override returns (uint256) {
+        return balanceOfWant().add(balanceOfPool());
+    }
+
+    /// @notice This method is defined in the BaseStrategy contract and is meant
+    ///         to provide an accurate conversion from amtInWei (denominated in wei)
+    ///         to want token (using the native decimal characteristics of want token).
+    /// @param amtInWei The amount (in wei/1e-18 ETH) to convert to want tokens.
+    /// @return The amount in want tokens.
+    function ethToWant(uint256 amtInWei)
+        public
+        view
+        virtual
+        override
+        returns (uint256)
+    {
+        // TODO: Create an accurate price oracle.
+        return amtInWei;
+    }
+
+    /// @notice This method is defined in the BaseStrategy contract and is
+    ///         meant to perform any adjustments to the core position(s) of this
+    ///         strategy, given what change the vault made in the investable
+    ///         capital available to the strategy. All free capital in the
+    ///         strategy after the report was made is available for reinvestment.
+    ///         This strategy implements the aforementioned behavior by taking
+    ///         its balance of the vault's underlying token and depositing it to
+    ///         the tBTC v2 Saddle reward pool.
+    /// @param debtOutstanding Will be 0 if the strategy is not past the
+    ///        configured debt limit, otherwise its value will be how far past
+    ///        the debt limit the strategy is. The strategy's debt limit is
+    ///        configured in the vault.
+    function adjustPosition(uint256 debtOutstanding) internal override {
+        uint256 wantBalance = balanceOfWant();
+        if (wantBalance > 0) {
+            want.safeIncreaseAllowance(lpRewardsTbtcV2Saddle, wantBalance);
+            ILPRewardsTBTCV2Saddle(lpRewardsTbtcV2Saddle).stake(wantBalance);
+        }
+    }
+
+    /// @notice Withdraws a portion of the vault's underlying token from
+    ///         the Saddle reward pool.
+    /// @param amount Amount to withdraw.
+    /// @return Amount withdrawn.
+    function withdrawSome(uint256 amount) internal returns (uint256) {
+        amount = Math.min(amount, balanceOfPool());
+        uint256 initialWantBalance = balanceOfWant();
+        // Withdraw some vault's underlying tokens but do not claim the rewards
+        // accumulated so far.
```

Comment:
Outdated comment.

Reply:
I removed the comment https://github.com/keep-network/tbtc-v2/pull/13/commits/5267c11fefbf0a07b3e4b19a6af70b943e735bf6. The code seems to be rather self-explanatory.